src/rust/engine/rule_graph/src/builder.rs

```diff
@@ -416,26 +447,43 @@ impl<'t, R: Rule> Builder<'t, R> {
       } else {
         continue;
       };
-      if !matches!(node.node, Node::Rule(_)) {
-        continue;
+      match node.node {
```

Comment:
Nit: it would be nice to put the fall-through case at the bottom of the match. I also prefer to have an empty match statement return a `()` rather than an empty `{}` block, since the former is less verbose.